circuits/statetransition/merkleproof.go

```diff
@@ -219,3 +219,19 @@ func padStateSiblings(unpackedSiblings []*big.Int) [types.CensusTreeMaxLevels]fr
 	}
 	return paddedSiblings
 }
+
+// TruncateKeyForArbo helper function truncates a key to a given size to be
+// compared with the key included in an arbo proof. It only matches if the
+// tree was generated with the BigInt API of arbo. It converts the key to
+// bytes, swaps the endianness, crops it to the given size, and then swaps
+// the endianness back. It returns the truncated key as a frontend.Variable.
+func TruncateKeyForArbo(api frontend.API, input frontend.Variable, size int) (frontend.Variable, error) {
```

Comment:
I'd not mention Arbo here, since it is a specific implementation of the iden3 SMT. Maybe just `TruncateMerkleProofKey`